### PR TITLE
Adding extented hugepages 2Mi TC

### PR DESCRIPTION
**Pull request by edcdavid**
This extended test case adds a check to verify that every pod using hugepages is using only the 2Mi size
Sample pass/fail pods are at:
https://github.com/test-network-function/cnf-certification-test-partner/pull/169

### CODE REVIEW

File: cnf-certification-test/platform/suite.go [hunk 1]
```diff
@@ -565,3 +573,20 @@ func testNodeOperatingSystemStatus(env *provider.TestEnvironment) {
...
 }
+
+func testPodHugePages2M(env *provider.TestEnvironment) {
+	var badPods, errorPods []*provider.Pod
+	for _, put := range env.HugepagesPods {
+		result, err := put.CheckResourceOnly2MiHugePages()
+		if err != nil {
+			tnf.ClaimFilePrintf("Could not check hugepages because of err: %s", err)
+			errorPods = append(errorPods, put)
+			continue
```
Comment (sebrandon1) [inline cnf-certification-test/platform/suite.go:584]: `CheckResourceOnly2MiHugePages` doesn't return an error, this section might not be needed?

**Review by sebrandon1 — COMMENTED** [review]
Some comments!

File: pkg/provider/hugepages.go [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+				return false, nil
+			}
+		}
+		for name := range cut.Resources.Limits {
+			if strings.Contains(name.String(), hugePages) && name != hugePages2Mi {
+				return false, nil
+			}
+		}
+	}
+	return true, nil
```
[the remark this answers is not in the document]
Comment (edcdavid) [reply pkg/provider/hugepages.go]: fixed

File: cnf-certification-test/platform/suite.go (hunk 1 above)
Comment (edcdavid) [reply cnf-certification-test/platform/suite.go]: fixed

File: pkg/provider/hugepages.go [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License along
+// with this program; if not, write to the Free Software Foundation, Inc.,
+// 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+package provider
```
[the remark this answers is not in the document]
Comment (edcdavid) [reply pkg/provider/hugepages.go]: fixed

File: go.sum [hunk 4]
```diff
@@ -1620,6 +1620,8 @@ k8s.io/apimachinery v0.19.0/go.mod h1:DnPGDnARWFvYa3pMHgSxtbZb7gpzzAZ1pTfaUNDVlm
 k8s.io/apimachinery v0.23.0/go.mod h1:fFCTTBKvKcwTPFzjlcxp91uPFZr+JA0FubU4fLzzFYc=
 k8s.io/apimachinery v0.24.2/go.mod h1:82Bi4sCzVBdpYjyI4jY6aHX+YCUchUIrZrXKedjd2UM=
 k8s.io/apimachinery v0.24.3/go.mod h1:82Bi4sCzVBdpYjyI4jY6aHX+YCUchUIrZrXKedjd2UM=
+k8s.io/apimachinery v0.25.0 h1:MlP0r6+3XbkUG2itd6vp3oxbtdQLQI94fD5gCS+gnoU=
```
Comment (sebrandon1) [inline go.sum:1623]: This might need a rebase from #520

**Review by sebrandon1 — APPROVED** [review]
Approved.  One nit about the go.sum.

File: go.sum (hunk 4 above)
Comment (edcdavid) [reply go.sum:1620]: rebased

**Comment by ramperher** [comment]
I'll add `build-depends: 26152` to the description of this PR to try to test a hugepage definition in our workload, to check if this test is going fine

**Review by greyerof — APPROVED** [review]

**Comment by ramperher** [comment]
> * FAILURE https://www.distributed-ci.io/jobs/9bce3c5b-5b1f-43ba-b389-770105fc88e0/jobStates
>   from [Adding extented hugepages 2Mi TC #517](https://github.com/test-network-function/cnf-certification-test/pull/517)

It is not because of your change, you can go ahead with it

**Pull request merged**